### ISSUE TITLE
Clean up XML Schema XSD

### ISSUE THREAD

**Issue by goneall** [spdx/license-list-XML#890]
The current XML schema is a bit hard to follow.  For example, there are two definitions of the `<optional>` element which cost some extra time in creating pull request #889 

There is also some inconsistencies in the level of documentation.

**Comment by zvr** [comment]
I'm willing to work on this.
Oldie but goodie resource: http://www.xfront.com/BestPracticesHomepage.html

**Comment by swinslow** [comment]
Thank you @zvr!

**Comment by swinslow** [comment]
Copied over from initially-separate issue in #897: 

> Add the attribute spacing to the alt element
> 
> Similar spacing issues can occur with the <alt elements as what happened to the <optional> elements (see PR #896 for reference).

@zvr indicated that this would be included as part of #890, so I'm combining the issues here.

@zvr, no rush at all :) but wanted to see if you expect that #890 will be part of the 3.8 release, or if I should mark it as part of a later release.

**Comment by zvr** [comment]
Sorry for not closing this yet... I have spent time on it, but it still needs more work. Definitely not for 3.8; I've marked it for later release.

Actually, I was thinking that it might be worthwhile for the schema and its documentation (#898) to be separated in another repo, so we can work on it independently... It might be useful to have its own ticket queue (e.g. the spacing attribute or the valid attribute for links).

I expect that the clean-up might introduce some non-backwards compatible changes -- but the actual data of all licenses can be easily automatically transformed to the new structure, once we all agree and publish it.

What do you all think?

**Comment by goneall** [comment]
> I was thinking that it might be worthwhile for the schema and its documentation (#898) to be separated in another repo, so we can work on it independently

We could separate it into its own repo, or we could add a tag to all the issues and PR's associated with the schema.  

> I expect that the clean-up might introduce some non-backwards compatible changes -- but the actual data of all licenses can be easily automatically transformed to the new structure, once we all agree and publish it.

This would be a good time to make these breaking changes since we haven't standardized and published the schema for external use.  Perhaps after the clean up we could publish this as a standard format for the licenses we publish out to the licenseListData.  I may want to suggest a few breaking changes myself.

**Comment by swinslow** [comment]
Thanks @zvr! About moving it to a separate repo, I don't have a strong feeling either way.

It would be great to be able to isolate out the "pure legal team" issues (e.g. new license requests, etc.) from those that are primarily technical. I suppose that could be either through fully-separate repos or else perhaps through more judicious use of issue labels, etc.

I'll defer to you and @goneall and others on whether there's any technical benefit to having the schema definition living in the same repo as the XML files themselves. I am not knowledgeable enough to have useful input there.  =)

**Comment by jlovejoy** [comment]
closing this and consolidating into #1369
